### PR TITLE
Copyedit and organize decK best practices documentation #3247

### PR DESCRIPTION
**Pull request by Rutam21**
### Summary
This PR refactors the decK best practices documentation. 

### Reason
Resolves the issue - https://github.com/Kong/docs.konghq.com/issues/3247

### Testing
The page preview looks good.

### CODE REVIEW

File: app/deck/1.8.x/guides/best-practices.md [hunk 1]
```diff
@@ -4,41 +4,39 @@ toc: false
...
 - Always ensure that you have one decK process running at any time. Multiple
-  process will step on each other and can corrupt Kong's configuration.
+  processes step on each other and can corrupt Kong's configuration.
 - Do not mix up decK's declarative configuration with `cURL` or any other
   script. Either manage the configuration with decK or manage it with your
-  homegrown script. Mixing the two on the same dataset will get cumbersome
-  and will be error-prone.
-- If you have a very large installation, then it is recommended to split out
+  homegrown script. Mixing the two on the same dataset gets cumbersome
+  and is error-prone.
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:11]:
```suggestion
  homegrown script. Mixing the two on the same dataset is cumbersome and error-prone.
```
Simplifying the phrasing a bit more.

File: app/deck/1.8.x/guides/best-practices.md [hunk 2]
```diff
@@ -4,41 +4,39 @@ toc: false
...
+- If you have a very large installation, then you can split out
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:12]:
```suggestion
- If you have a very large installation, you can split out
```

File: app/deck/1.8.x/guides/best-practices.md [hunk 3]
```diff
@@ -4,41 +4,39 @@ toc: false
...
   your configuration into smaller subsets. You can find more info for it
   in the guide to practicing
   [distributed configuration](/deck/{{page.kong_version}}/guides/distributed-configuration).
 - Always use a pinned version of decK and Kong.
-  Achieving declarative configuration is
-  not easy because details matter a lot. Use a specific version of decK in
-  production. If you're going to start using a new version of decK or Kong,
+  Use a specific version of decK in
+  production to achieve declarative configuration. If you're going to opt for a new version of decK or Kong,
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:18]:
```suggestion
  Use a specific version of decK in production to achieve declarative 
  configuration. If you upgrade to a new version of decK or Kong,
```

File: app/deck/1.8.x/guides/best-practices.md [hunk 4]
```diff
@@ -4,41 +4,39 @@ toc: false
...
-  this file as it can have a huge impact on your security.
-  You should store these in an encrypted form and provide a plaintext version
-  of this on a need-by-need basis.
-- If you have a very large number of consumers in your database, do not export
-  or manage them using decK. Declarative configuration is for ... configuration,
-  it is not meant for end-user data, which can easily grow into hundreds of
-  thousands or millions.
+- decK does not manage encryption of sensitive information. The state file stores the private
+  keys of your certificates and credentials of consumers in
+  plaintext. Please be careful in how and where you store
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:22]:
```suggestion
- decK does not manage encryption of sensitive information. The state file 
  stores the private keys of your certificates and credentials of consumers in
  plaintext. Be careful in how and where you store
```

File: app/deck/1.8.x/guides/best-practices.md [hunk 5]
```diff
@@ -4,41 +4,39 @@ toc: false
...
+  this file to avoid any security breaches.
+  You should always store the sensitive information in an encrypted form and provide a plaintext version
+  of it on a need-only basis.
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:25]:
```suggestion
  Always store the sensitive information in an encrypted form and provide a plaintext 
  version of it on a need-only basis.
```

File: app/deck/1.8.x/guides/best-practices.md [hunk 6]
```diff
@@ -4,41 +4,39 @@ toc: false
...
+- If you have many consumers in your database, do not export
+  or manage them using decK. Declarative configuration is only for configuration.
+  It is not meant for the end-user data which can easily grow into hundreds of thousands or millions.
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:28]:
```suggestion
- If you have many consumers in your database, do not export
  or manage them using decK. Declarative configuration is only meant for entity 
  configuration. It is not meant for end-user data, which can easily grow into 
  hundreds of thousands or millions of records.
```

File: app/deck/1.8.x/guides/best-practices.md [hunk 7]
```diff
@@ -4,41 +4,39 @@ toc: false
...
 - Always run a `deck diff` command before running a `deck sync`
-  to ensure the change that is taking place.
+  to ensure that the change works properly.
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:30]:
```suggestion
  to ensure that the change is correct.
```

File: app/deck/1.8.x/guides/best-practices.md [hunk 8]
```diff
@@ -4,41 +4,39 @@ toc: false
...
 - Adopt a [CI-driven configuration](/deck/{{page.kong_version}}/guides/ci-driven-configuration) practice.
-- Always secure Kong's Admin API with some kind of authentication method.
-- Do not write the state file by hand, it will be very error-prone.
-  Instead, use Kong's Admin API to
-  configure Kong for the first time, then export the configuration. Any
+- Always secure Kong's Admin API with a reliable authentication method.
+- Do not write the state file by hand to avoid errors.
+  Always use Kong's Admin API to
+  configure Kong for the first time and then export the configuration. Any
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:35]:
```suggestion
  Use Kong's Admin API to configure Kong for the first time, then 
  export the configuration to a declarative configuration file. Any
```

File: app/deck/1.8.x/guides/best-practices.md [hunk 9]
```diff
@@ -4,41 +4,39 @@ toc: false
...
   subsequent changes should be made by manually editing the file and pushing
-  the change via CI. If you're making a large change, make the change in Kong,
-  export the new file, and then diff the two state files to review the changes
+  the change via CI. If you're making a larger change, make the changes in Kong and
+  export the new file. Then you can differentiate the two state files to review the changes
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:38]:
```suggestion
  export the new file. Then you can `diff` the two state files to review the changes
```
diff is fine to use here, it's the specific command name.

File: app/deck/1.8.x/guides/best-practices.md [hunk 10]
```diff
@@ -4,41 +4,39 @@ toc: false
...
-  Instead, use Kong's Admin API to
-  configure Kong for the first time, then export the configuration. Any
+- Always secure Kong's Admin API with a reliable authentication method.
+- Do not write the state file by hand to avoid errors.
+  Always use Kong's Admin API to
+  configure Kong for the first time and then export the configuration. Any
   subsequent changes should be made by manually editing the file and pushing
-  the change via CI. If you're making a large change, make the change in Kong,
-  export the new file, and then diff the two state files to review the changes
+  the change via CI. If you're making a larger change, make the changes in Kong and
```
Comment (lena-larionova) [inline app/deck/1.8.x/guides/best-practices.md:37]:
```suggestion
  the change via CI. If you're making a larger change, make the change in Kong first, then
```

**Review by lena-larionova — COMMENTED** [review]
Thanks for the contribution! I left some feedback for style and technical accuracy.

File: app/deck/1.8.x/guides/best-practices.md (hunk 9 above)
Comment (Rutam21) [reply app/deck/1.8.x/guides/best-practices.md:38]: Okay got it.

**Comment by Rutam21** [comment]
Thanks @lena-larionova for your time to review and provide feedback. All the suggested changes have been incorporated. Let me know if it looks good to you now.

**Review by lena-larionova — APPROVED** [review]
Thanks for implementing all the feedback! LGTM :shipit:

**Pull request merged**